src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SvType.java

```diff
@@ -0,0 +1,155 @@
+
+        @Override
+        public String toString() {
+            return TYPES.DEL.name();
+        }
+
+        @SuppressWarnings("unchecked")
+        Deletion(final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations) {
+            super(getIDString(novelAdjacencyReferenceLocations),
+                    Allele.create(SVConstants.CallingStepConstants.VCF_ALT_ALLELE_STRING_DEL),
+                    -(novelAdjacencyReferenceLocations.leftJustifiedRightRefLoc.getStart() - novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getEnd()),
+                    novelAdjacencyReferenceLocations.complication.dupSeqForwardStrandRep.isEmpty() ? Collections.EMPTY_MAP : Collections.singletonMap(SVConstants.CallingStepConstants.TANDUP_CONTRACTION_STRING, ""));
+        }
+
+        private static String getIDString(final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations) {
+            final String contig = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getContig();
+            final int start = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getEnd();
+            final int end = novelAdjacencyReferenceLocations.leftJustifiedRightRefLoc.getStart();
+            return  ((novelAdjacencyReferenceLocations.complication.dupSeqForwardStrandRep.isEmpty()) ? TYPES.DEL.name() : TANDEMUPLICATION_CONTRACTION_ID_START_STRING)
+                    + SVConstants.CallingStepConstants.VARIANT_ID_FIELD_SEPARATOR +
+                    contig + SVConstants.CallingStepConstants.VARIANT_ID_FIELD_SEPARATOR + start + SVConstants.CallingStepConstants.VARIANT_ID_FIELD_SEPARATOR + end;
+        }
+    }
+
+    static final class Insertion extends SvType {
+
+        @Override
+        public String toString() {
+            return TYPES.INS.name();
+        }
+
+        @SuppressWarnings("unchecked")
+        Insertion(final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations) {
+            super(getIDString(novelAdjacencyReferenceLocations),
+                    Allele.create(SVConstants.CallingStepConstants.VCF_ALT_ALLELE_STRING_INS),
+                    novelAdjacencyReferenceLocations.complication.insertedSequenceForwardStrandRep.length(),
+                    Collections.EMPTY_MAP);
+        }
+
+        private static String getIDString(final NovelAdjacencyReferenceLocations novelAdjacencyReferenceLocations) {
+            final String contig = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getContig();
+            final int start = novelAdjacencyReferenceLocations.leftJustifiedLeftRefLoc.getEnd();
+            final int end = novelAdjacencyReferenceLocations.leftJustifiedRightRefLoc.getStart();
+
+            final String startString;
+            if (novelAdjacencyReferenceLocations.complication.dupSeqForwardStrandRep.isEmpty()) {
+                startString = TYPES.INS.name();
+            } else {
+                startString = "DuplicationTandem" +
```

Comment:
not sure why i didn't catch this. it is now changed so that tandem duplication is handled completely in the `DuplicationTandem` class.